### DISCUSSION TITLE
Events from fields containing Array

### DISCUSSION THREAD

**Discussion by muhammadsaeedparacha** [wdelfuego/nova-calendar#11] [Ideas & Feature Requests]
I have a json field that contains an array, each array item has a date that I want to show on the calendar.

I'm creating the functionality and will send a PR.

**Comment by wdelfuego** [comment]
I've turned the issue into a Discussion since I consider this a feature request; I'll use issues exclusively for bugs and confirmed to do's (roadmap). Feel free to create Discussions for features you think should be on the roadmap.

👍 Looking forward to your PR!

**Comment by muhammadsaeedparacha** [comment]
Feature: Accept columns that have values: array, json, or custom cast objects.

Tech Plan:

```
public function novaResources() : array
    {
        return [
            User::class => 'created_at',
            Activity::class => [
                'type' => CalendarDataProvider::JSON,
                'column' => 'events'
                'value' => 'activity->activity_at'
            ]
        ];
    }
```

Assuming `Activity` model has a `events` column that has an array of json objects or json keys. like:
```
$activity->events = [
    ['activity' => ['activity_at' => '2022-06-05 12:30:00']],
    ['activity' => ['activity_at' => '2022-06-04 12:30:00']],
]
```

This might change if I bump into issues or get new insights

**Comment by wdelfuego** [comment]
I like the feature but won't merge the suggested implementation; I prefer a more flexible approach; for example by adding something like a `CalendarDataProvider::resourceToEvents` method and/or allowing the Nova resource itself to supply a `toEvents` method. That way any Nova Resource can be freely converted into multiple events without cluttering up  the calendar data provider.

**Reply by muhammadsaeedparacha** [reply]
Valid point, I like the keeping it clean approach

**Comment by wdelfuego** [comment]
Version 1.6 has just been released; this update allows you to implement custom EventGenerators which allow you to implement your own custom mappings from Nova resources to calendar events.

Take a look at the [documentation](https://wdelfuego.github.io/nova-calendar/custom-event-generators.html) and let me know if you run into any issues!
